### PR TITLE
Add CODEOWNERS

### PR DESCRIPTION
**Pull request by stephengeller**
## What does this change?

https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

Following the above docs, we can create a CODEOWNERS file to automatically ensure the correct team is notified on incoming pull requests!

## Does this change require changes to existing projects or CDK CLI?

Nope.

## How to test

Create a PR (after this one) and see the team correctly tagged as a review request. If this doesn't work, we may need to prepend the name as `@guardian/developer-experience`.

## How can we measure success?

The DevX team is automatically requested for review for PRs in this repo.

### CODE REVIEW

**Comment by akash1810** [comment]
From https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax:

> The pattern is followed by one or more GitHub usernames or team names using the standard @username or @org/team-name format.


So I think it should be `@guardian/developer-experience`.

**Comment by sihil** [comment]
It's also missing a pattern on the LHS. I imagine `*` would suffice.

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+@developer-experience
```
Comment (stephengeller) [inline CODEOWNERS:1]:
```suggestion
* @guardian/developer-experience
```

**Comment by stephengeller** [comment]
Doh, thank you @akash1810 and @sihil , that's been updated.

**Review by akash1810 — APPROVED** [review]

**Review by sihil — APPROVED** [review]

**Pull request merged**